File: C++/532. K-diff Pairs in an Array.cpp
class Solution {
public:
    int findPairs(vector<int>& nums, int k) {
        if(k < 0)
            return 0;
        unordered_map<int, int> maple;
        for(auto num: nums)
            maple[num] += 1;
        int ans = 0;
        for(auto &it: maple)
        {
            if(k==0 && it.second>1)
                ans += 1;
            else if(k != 0 && maple.find(it.first + k) != maple.end())
                ans += 1;
        }
        return ans;
    }
};

File: Python/532. K-diff Pairs in an Array.py
class Solution:
    def findPairs(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        if k < 0:
            return 0
        maple = dict()
        for num in nums:
            maple[num] = maple.get(num, 0) + 1
        ans = 0
        for num in maple:
            if k == 0 and maple[num] > 1:
                ans += 1
            elif k != 0:
                tmp = num + k
                if maple.get(tmp) != None:
                    ans += 1
        return ans